components/identity/org.wso2.carbon.identity.totp.valve/src/main/java/org/wso2/carbon/identity/totp/valve/BasicAuthHandler.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.totp.valve;
+
+import org.apache.axiom.om.util.Base64;
+import org.apache.catalina.connector.Request;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.context.PrivilegedCarbonContext;
+import org.wso2.carbon.user.api.UserRealm;
+import org.wso2.carbon.user.api.UserStoreException;
+import org.wso2.carbon.user.core.service.RealmService;
+import org.wso2.carbon.utils.multitenancy.MultitenantConstants;
+import org.wso2.carbon.utils.multitenancy.MultitenantUtils;
+
+import java.nio.charset.Charset;
+import java.util.Map;
+
+/**
+ * Basic Authentication handler class.
+ */
+public class BasicAuthHandler implements TOTPAuthenticationHandler {
+	private static Log log = LogFactory.getLog(BasicAuthHandler.class);
+	private int priority;
+	private Map<String, String> properties;
+	private final int DEFAULT_PRIORITY = 5;
+
+	@Override
+	public boolean canHandler(Request request) {
+
+		String authheader = request.getHeader(Constants.AUTHORIZATION_HEADER);
+		if (authheader != null && authheader.startsWith(Constants.BASIC_AUTH_HEADER)) {
+			return true;
+		}
+		return false;
```

Comment:
Can be simplified as "return (authheader != null && authheader.startsWith(Constants.BASIC_AUTH_HEADER);"